mixer/pkg/runtime2/routing/table.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package routing
+
+import (
+	tpb "istio.io/api/mixer/v1/template"
+	"istio.io/istio/mixer/pkg/adapter"
+	"istio.io/istio/mixer/pkg/attribute"
+	"istio.io/istio/mixer/pkg/il/compiled"
+	"istio.io/istio/mixer/pkg/runtime2/config"
+	"istio.io/istio/mixer/pkg/template"
+	"istio.io/istio/pkg/log"
+)
+
+// Table is the main routing table. It is used to find the set of handlers that should be invoked, along with the
+// instance builders and match conditions.
+type Table struct {
+
+	// ID of this table. This is based on the config snapshot ID. IDs are unique within the life-span of a Mixer instance.
+	ID int64
```

Comment:
Should this be exposed? Or do we want some accessor, like `func (t Table) ID() int64`?